server/types/serialization.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+type serializationID byte
+
+const (
+	serializationID_Bool      serializationID = 1
+	serializationID_BoolArray serializationID = 2
+)
+
+// init registers the serialization and deserialization functions.
+func init() {
+	types.RegisterExtendedTypeSerializers(SerializeType, DeserializeType)
+}
+
+// SerializeType is able to serialize the given extended type into a byte slice. All extended types will be defined
+// by DoltgreSQL.
+func SerializeType(extendedType types.ExtendedType) ([]byte, error) {
+	switch extendedType.(type) {
+	case BoolType:
+		return []byte{byte(serializationID_Bool)}, nil
```

Comment:
Probably nice to have something more legible like this (type string e.g.) since this is going to get appended with "custom_".

"custom_1" doesn't tell me anything very useful in the debugger